File: console-v2/ui/src/welcome/module.ts
/*******************************************************************************
* Copyright (c) 2011, 2016 Eurotech and/or its affiliates                       
*                                                                               
* All rights reserved. This program and the accompanying materials              
* are made available under the terms of the Eclipse Public License v1.0         
* which accompanies this distribution, and is available at                      
* http://www.eclipse.org/legal/epl-v10.html                                     
*                                                                               
* Contributors:                                                                 
*     Eurotech - initial API and implementation                                 
*                                                                               
*******************************************************************************/
import WelcomeCtrl from "./controllers/WelcomeCtrl";

angular.module("app.welcome", ["ui.router"])
    .config(["$stateProvider", ($stateProvider: angular.ui.IStateProvider) => {
        $stateProvider.state("kapua.welcome", {
            views: {
                "kapuaView@kapua": {
                    template: require("./views/welcome.html"),
                    controller: "WelcomeCtrl as vm"
                }
            },
            url: "/welcome"
        });
    }])
    .controller("WelcomeCtrl", WelcomeCtrl);

File: client/gateway/spi/src/main/java/org/eclipse/kapua/client/gateway/spi/Module.java
/*******************************************************************************
 * Copyright (c) 2017, 2022 Red Hat Inc and others.
 *
 * This program and the accompanying materials are made
 * available under the terms of the Eclipse Public License 2.0
 * which is available at https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *     Red Hat Inc - initial API and implementation
 *******************************************************************************/
package org.eclipse.kapua.client.gateway.spi;

public interface Module {

    public default void initialize(final ModuleContext context) {
    }

    public default void applicationAdded(final String applicationId) {
    }

    public default void applicationRemoved(final String applicationId) {
    }

    public default void connected() {
    }

    public default void disconnected() {
    }
}
